share/availability/discovery/options.go

```diff
@@ -0,0 +1,69 @@
+package discovery
+
+import (
+	"fmt"
+	"time"
+)
+
+// Parameters is the set of Parameters that must be configured for the Discovery module
+type Parameters struct {
+	// PeersLimit defines the soft limit of FNs to connect to via discovery.
+	// Set 0 to disable.
+	PeersLimit uint
+	// AdvertiseInterval is a interval between advertising sessions.
+	// Set -1 to disable.
+	// NOTE: only full and bridge can advertise themselves.
+	AdvertiseInterval time.Duration
+	// discoveryRetryTimeout is an interval between discovery attempts
+	// when we discovered lower than PeersLimit peers.
+	discoveryRetryTimeout time.Duration
+}
+
+// Option is a function that configures Discovery Parameters
+type Option func(*Parameters)
```

Comment:
What we really need functional options for config values that are exported and could be set directly?

Reply:
Unclear question